File: app/src/main/java/me/sargunvohra/android/diningcourts/main/MainActivity.java
package me.sargunvohra.android.diningcourts.main;

import android.graphics.Color;
import android.support.annotation.ColorRes;
import android.support.annotation.IdRes;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;

import com.roughike.bottombar.BottomBar;
import com.roughike.bottombar.OnTabSelectListener;

import me.sargunvohra.android.diningcourts.R;

public class MainActivity extends AppCompatActivity implements OnTabSelectListener {

    private ToolbarColorAnimator toolbarColorAnimator;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        toolbarColorAnimator = new ToolbarColorAnimator(toolbar, Color.BLACK);

        BottomBar bottomBar = (BottomBar) findViewById(R.id.bottomBar);
        bottomBar.setOnTabSelectListener(this);
    }

    @Override
    public void onTabSelected(@IdRes int tabId) {
        @ColorRes int color;
        switch (tabId) {
            case R.id.tab_dining:
                color = R.color.primary_dining;
                break;
            case R.id.tab_restaurants:
                color = R.color.primary_restaurants;
                break;
            case R.id.tab_settings:
                color = R.color.primary_settings;
                break;
            default:
                throw new IllegalStateException("Unhandled bottom bar tab!");
        }
        toolbarColorAnimator.animateToColor(ContextCompat.getColor(MainActivity.this, color));
    }
}

File: app/src/main/java/me/sargunvohra/android/diningcourts/main/MainActivity.kt
package me.sargunvohra.android.diningcourts.main

import android.graphics.Typeface
import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v4.view.ViewPager
import android.support.v7.widget.LinearLayoutManager
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import com.karumi.rosie.renderer.RosieRenderer
import com.pedrogomez.renderers.ListAdapteeCollection
import com.pedrogomez.renderers.RVRendererAdapter
import com.pedrogomez.renderers.Renderer
import com.pedrogomez.renderers.RendererBuilder
import kotlinx.android.synthetic.main.activity_main.*
import me.sargunvohra.android.diningcourts.R
import me.sargunvohra.android.diningcourts.base.BaseActivity
import me.sargunvohra.android.diningcourts.config.dagger.MainModule
import me.sargunvohra.lib.ktunits.days
import org.jetbrains.anko.*
import java.text.SimpleDateFormat
import java.util.*

class MainActivity : BaseActivity() {

    override fun getLayoutId() = R.layout.activity_main

    override fun getActivityScopeModules() = listOf(MainModule())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // setup action bar
        setSupportActionBar(toolbar)

        // setup view pager
        viewPager.adapter = MainPagerAdapter(supportFragmentManager, Date())
        tabLayout.setupWithViewPager(viewPager)

        // setup fab toolbar
        fabToolbar.setFab(fab)
        fab.onClick {
            fabToolbar.expandFab()
        }

        // setup fab toolbar with view pager
        viewPager.addOnPageChangeListener(object : ViewPager.SimpleOnPageChangeListener() {
            override fun onPageSelected(position: Int) = fabToolbar.slideInFab()
        })
    }
}